src/openforms/payments/contrib/worldline/plugin.py

```diff
@@ -0,0 +1,255 @@
+from collections.abc import Generator
+from typing import Any
+
+from django.http import (
+    HttpRequest,
+    HttpResponse,
+    HttpResponseBadRequest,
+    HttpResponseRedirect,
+    HttpResponseServerError,
+)
+from django.urls import reverse
+from django.utils.translation import gettext_lazy as _
+
+import structlog
+from onlinepayments.sdk.api_exception import ApiException
+from onlinepayments.sdk.communicator import CommunicationException
+from onlinepayments.sdk.domain.create_hosted_checkout_request import (
+    CreateHostedCheckoutRequest,
+)
+from rest_framework import serializers
+from rest_framework.exceptions import APIException
+from rest_framework.request import Request
+
+from openforms.api.fields import PrimaryKeyRelatedAsChoicesField
+from openforms.config.data import Entry
+from openforms.frontend.frontend import get_frontend_redirect_url
+from openforms.payments.base import BasePlugin, PaymentInfo
+from openforms.payments.constants import (
+    PAYMENT_STATUS_FINAL,
+    PaymentRequestType,
+    PaymentStatus,
+)
+from openforms.payments.models import SubmissionPayment
+from openforms.submissions.tokens import submission_status_token_generator
+from openforms.utils.mixins import JsonSchemaSerializerMixin
+
+from ...registry import register
+from .models import WorldlineMerchant
+from .typing import (
+    AmountOfMoney,
+    CheckoutInput,
+    Order,
+    PaymentOptions,
+)
+from .utils import get_payment_status
+
+logger = structlog.stdlib.get_logger(__name__)
+
+
+class WorldlineOptionsSerializer(JsonSchemaSerializerMixin, serializers.Serializer):
+    merchant = PrimaryKeyRelatedAsChoicesField(
```

Comment:
since this is new code, can we make it export/setup-configuration friendly by not relying on PK as identifier, but instead a unique `identifier` slug field? See also the ticket Anna is working on: #5384 - it's the same root problem.

Reply:
@annashamray FYI - make sure to talk to Sonny if/when you make a new generic API serializer field or modify the existing one.

Probably it's a good idea to deprecate `PrimaryKeyRelatedAsChoicesField` in favour of the alternative.